src/vos/vos_dtx.c

```diff
@@ -4006,3 +4006,28 @@ vos_dtx_local_end(struct dtx_handle *dth, int result)
 
 	return result;
 }
+
+uint32_t
+vos_dtx_get_cmt_cnt(daos_handle_t coh)
+{
+	struct umem_instance   *umm;
+	struct vos_container   *cont;
+	struct vos_dtx_blob_df *dbd;
+	uint32_t                cnt;
+
+	cont = vos_hdl2cont(coh);
+	D_ASSERT(cont != NULL);
+
+	cnt = 0;
+	umm = vos_cont2umm(cont);
+	dbd = umem_off2ptr(umm, cont->vc_cont_df->cd_dtx_committed_head);
+	while (dbd != NULL) {
+		D_ASSERTF(dbd->dbd_magic == DTX_CMT_BLOB_MAGIC,
+			  "Corrupted committed DTX blob (2) %x\n", dbd->dbd_magic);
```

Comment:
Different from regular `vos` logic, `ddb` is usually to handle system corruption, then it is better not assertion the on-disk data, instead, we can report some error.

Reply:
> * [x]  Change assert into error code.

Fixed with commit 49be49554